editor/lang/treeNode.cc

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "treeNode.hh"
+
+void TreeNode::add_child(TreeNode child) {
+  this->m_children.push_back(new TreeNode(child.m_token_node));
+  int size = this->m_children.size();
+
+  // Adding children of the incoming node if exist.
+  for (int i = 0; i < child.m_children.size(); ++i) {
```

Comment:
I feel like you could've avoided this if you took a pointer as the parameter instead.
Same with the siblings below.